### PR TITLE
OperatingSystemView: display PRETTY_NAME when possible

### PR DESCRIPTION
**Pull request by bobby285271**
Implementation of https://github.com/NixOS/nixpkgs/pull/130380#issuecomment-885147633.

`VERSION` in NixOS is a bit too long to display in the latest switchboard:

```
NAME=NixOS
ID=nixos
VERSION="21.11.git.165c18ade65 (Porcupine)"
VERSION_CODENAME=porcupine
VERSION_ID="21.11.git.165c18ade65"
PRETTY_NAME="NixOS 21.11 (Porcupine)"
LOGO="nix-snowflake"
HOME_URL="https://nixos.org/"
DOCUMENTATION_URL="https://nixos.org/learn.html"
SUPPORT_URL="https://nixos.org/community.html"
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
```

It will be great if we can display `PRETTY_NAME` when `NAME` is a prefix of it so we can still make `NAME` bold.
The currently used `NAME` + `VERSION` way can be used as an alternative if the requirements are not satisfied.
See below screenshot for reference.

Thanks in advance for reviewing :-)

### CODE REVIEW

File: src/Views/OperatingSystemView.vala [hunk 1]
```diff
@@ -68,13 +68,28 @@ public class About.OperatingSystemView : Gtk.Grid {
...
-            Environment.get_os_info (GLib.OsInfoKey.VERSION)
+            os_name,
+            os_version
         );
 
+        if (os_pretty_name != null
+            && os_name != null
+            && os_pretty_name.length > os_name.length
+            && os_pretty_name.slice(0, os_name.length) == os_name
+            && os_pretty_name.get(os_name.length) == ' ') {
```
Comment (danrabbit) [inline src/Views/OperatingSystemView.vala:86]: Wouldn't [`has_prefix`](https://valadoc.org/glib-2.0/string.has_prefix.html) work for this?
```suggestion
            && os_pretty_name.has_prefix (os_name + " ") {
```

File: src/Views/OperatingSystemView.vala [hunk 2]
```diff
@@ -68,13 +68,28 @@ public class About.OperatingSystemView : Gtk.Grid {
...
+            pretty_name = "<b>%s</b>%s".printf (
+                os_name,
+                os_pretty_name.slice(os_name.length, os_pretty_name.length)
```
Comment (danrabbit) [inline src/Views/OperatingSystemView.vala:89]: elementary code style is to use spaces before parentheses :) 
```suggestion
                os_pretty_name.slice (os_name.length, os_pretty_name.length)
```

**Review by danrabbit — CHANGES_REQUESTED** [review]
Hey right on, thanks for working on this! Just a couple of inline comments

**Comment by bobby285271** [comment]
> Hey right on, thanks for working on this! Just a couple of inline comments

Just did the update, please take a look, thanks!

**Comment by davidak** [comment]
This is not needed anymore for NixOS with https://github.com/NixOS/nixpkgs/pull/131967. The `VERSION` is now only the release number.

If it's not an issue with other distros, this can get closed.

**Comment by cassidyjames** [comment]
This looks clean enough and may be nice, but if it's not required for NixOS anymore, I'm also fine closing it. We can always re-open it or something like it if there's a specific downstream request.

**Pull request closed without merging**